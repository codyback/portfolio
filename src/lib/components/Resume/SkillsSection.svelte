<script lang="ts">
	import type { Skill } from '@prisma/client';

	export let skills: Skill[];
</script>

<section
	id="cvSkills"
	class="sm:col-span-1 mt-10 sm:mt-20"
>
	<div class="border-b-2 border-black pb-3">
		<h3 class="uppercase text-2xl font-bold text-right">Skills</h3>
	</div>

	<div class="border-t-4 border-black mt-0.5 pt-8">
		<p class="text-justify text-neutral-500">
			Skills include proficiency in multiple programming languages, frameworks,
			and tools, with experience in web, mobile and data analysis.
		</p>

		<div class="mt-6 font-bold flex-initial flex-wrap flex-row">
			{#each skills as skill (skill.id)}
				<span
					class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
				>
					{skill.name}
				</span>
			{/each}
		</div>
	</div>
</section>
