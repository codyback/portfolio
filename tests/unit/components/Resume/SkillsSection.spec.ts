import SkillsSection from '@/lib/components/Resume/SkillsSection.svelte';
import type { Skill } from '@prisma/client';
import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';

const skills: Skill[] = [
  {
    id: 'skill-1',
    name: 'skill 1',
    url: 'skill-1-url',
    homepageId: 'test',
  },
  {
    id: 'skill-2',
    name: 'skill 2',
    url: 'skill-2-url',
    homepageId: 'test',
  },
  {
    id: 'skill-3',
    name: 'skill 3',
    url: 'skill-3-url',
    homepageId: 'test',
  },
];

describe('CertificatesSection', () => {
  it('renders correctly with open false', () => {
    const { container } = render(SkillsSection, { skills });

    expect(container).toMatchSnapshot();
  });
});
