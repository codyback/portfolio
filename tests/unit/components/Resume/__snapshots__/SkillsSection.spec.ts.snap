// Vitest Snapshot v1

exports[`CertificatesSection > renders correctly with open false 1`] = `
<body>
  <div>
    <section
      class="sm:col-span-1 mt-10 sm:mt-20"
      id="cvSkills"
    >
      <div
        class="border-b-2 border-black dark:border-neutral-500 pb-3"
      >
        <h3
          class="uppercase text-2xl font-bold text-right"
        >
          Skills
        </h3>
      </div>
       
      <div
        class="border-t-4 border-black dark:border-neutral-500 mt-0.5"
      />
      <!--&lt;SectionHeading&gt;-->
       
      <div
        class="pt-8"
      >
        <p
          class="text-justify dark:text-neutral-300 text-neutral-500"
        >
          Skills include proficiency in multiple programming languages, frameworks,
			and tools, with experience in web, mobile and data analysis.
        </p>
         
        <div
          class="mt-6 font-bold flex-initial flex-wrap flex-row"
        >
          <span
            class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
          >
            skill 1
             
          </span>
          <span
            class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
          >
            skill 2
             
          </span>
          <span
            class="px-4 py-0.5 mx-1 border border-neutral-300 rounded-full text-xs uppercase"
          >
            skill 3
             
          </span>
        </div>
      </div>
    </section>
    <!--&lt;SkillsSection&gt;-->
  </div>
</body>
`;
